org.adempiere.base/src/org/compiere/model/PO_Record.java

```diff
@@ -59,46 +60,63 @@
 	/**
 	 * 	Delete Cascade including (selected)parent relationships
 	 *	@param AD_Table_ID table
-	 *	@param Record_ID record
+	 *	@param Record_IDorUU record ID (int) or UUID (String)
 	 *	@param trxName transaction
 	 *	@return false if could not be deleted
 	 */
-	static boolean deleteRecordIdCascade (int AD_Table_ID, int Record_ID, String trxName)
+	static boolean deleteRecordCascade (int AD_Table_ID, Serializable Record_IDorUU, String trxName)
 	{
-		KeyNamePair[] cascades = getTablesWithRecordIDColumn(MColumn.FKCONSTRAINTTYPE_ModelCascade, trxName);
+		int refId;
+		String columnName;
+		if (Record_IDorUU instanceof Integer) {
+			refId = DisplayType.RecordID;
+			columnName = "Record_ID";
+		} else if (Record_IDorUU instanceof String) {
+			refId = DisplayType.RecordUU;
+			columnName = "Record_UU";
+		} else {
+			log.warning(Record_IDorUU.getClass().getName() + " not supported for ID/UUID");
+			return true;
+		}
+		KeyNamePair[] cascades = getTablesWithConstraintType(refId, MColumn.FKCONSTRAINTTYPE_ModelCascade, trxName);
 		//	Table Loop
+		StringBuilder whereClause = new StringBuilder("AD_Table_ID=? AND ").append(columnName).append("=?");
 		for (KeyNamePair table : cascades)
 		{
 			//	DELETE FROM table WHERE AD_Table_ID=#1 AND Record_ID=#2
-			List<PO> poList = new Query(Env.getCtx(), table.getName(), "AD_Table_ID=? AND Record_ID=?", trxName)
-					.setParameters(AD_Table_ID, Record_ID)
+			List<PO> poList = new Query(Env.getCtx(), table.getName(), whereClause.toString(), trxName)
+					.setParameters(AD_Table_ID, Record_IDorUU)
 					.list();
 
 			int count = 0;
 			for(PO po : poList)
 			{
-				if (po.get_Table_ID() == AD_Table_ID && po.get_ID() == Record_ID)
+				if (   po.get_Table_ID() == AD_Table_ID
+					&& (   (Record_IDorUU instanceof Integer && po.get_ID() == (Integer)Record_IDorUU)
+						|| (Record_IDorUU instanceof String && Record_IDorUU.equals(po.get_UUID()))
+					   )
+					)
 					continue;
 				po.deleteEx(true);
 				count++;
 			}
 			if (count > 0)
-				if (log.isLoggable(Level.CONFIG)) log.config(table.getName() + " (" + AD_Table_ID + "/" + Record_ID + ") #" + count);
+				if (log.isLoggable(Level.CONFIG)) log.config(table.getName() + " (" + AD_Table_ID + "/" + Record_IDorUU + ") #" + count);
 		}
 		//	Parent Loop
 		for (int i = 0; i < s_parents.length; i++)
 		{
 			if (s_parents[i] == AD_Table_ID)
 			{
 				int AD_Table_IDchild = s_parentChilds[i];
+				StringBuilder whereClauseChild = new StringBuilder(" AD_Table_ID=? AND ").append(columnName).append(" IN (SELECT ")
+						.append(s_parentChildNames[i]).append("_ID FROM ")
+						.append(s_parentChildNames[i]).append(" WHERE ")
+						.append(s_parentNames[i]).append("_ID=?)");

```

Comment:
this is inconsistent. if the change is to support UUID only table, you can't assume parent is using _ID.

Reply:
I'm debugging this - it is a strange code anyways.

The deletion of orderlines used to happen on MOrder.beforeDelete - and I moved that to Model Cascade with ticket IDEMPIERE-2060

There is a bug as you noticed, but the code looks weird, I think is unnecessary.

Reply:
I've not inspect PO_Record before this, very strange one indeed.